src/service/resources-impl.js

```diff
@@ -1741,28 +1816,57 @@ export class Resource {
   }
 
   /**
-   * Calls element's documentInactiveCallback callback and resets state for
+   * Calls element's unlayoutCallback callback and resets state for
    * relayout in case document becomes active again.
    */
-  documentBecameInactive() {
-    if (this.state_ == ResourceState_.NOT_BUILT) {
+  unlayout() {
+    if (this.state_ != ResourceState_.LAYOUT_COMPLETE) {
```

Comment:
I think we should ignore `unlayoutCallback` for `NOT_BUILT` and `NOT_LAID_OUT`, but call it in all other cases.